paimon-core/src/main/java/org/apache/paimon/jdbc/JdbcUtils.java

```diff
@@ -0,0 +1,646 @@
+            "INSERT INTO "
+                    + DATABASE_PROPERTIES_TABLE_NAME
+                    + " ("
+                    + CATALOG_NAME
+                    + ", "
+                    + DATABASE_NAME
+                    + ", "
+                    + DATABASE_PROPERTY_KEY
+                    + ", "
+                    + DATABASE_PROPERTY_VALUE
+                    + ") VALUES ";
+    static final String INSERT_PROPERTIES_VALUES_BASE = "(?,?,?,?)";
+    static final String GET_ALL_DATABASE_PROPERTIES_SQL =
+            "SELECT * "
+                    + " FROM "
+                    + DATABASE_PROPERTIES_TABLE_NAME
+                    + " WHERE "
+                    + CATALOG_NAME
+                    + " = ? AND "
+                    + DATABASE_NAME
+                    + " = ? ";
+    static final String DELETE_DATABASE_PROPERTIES_SQL =
+            "DELETE FROM "
+                    + DATABASE_PROPERTIES_TABLE_NAME
+                    + " WHERE "
+                    + CATALOG_NAME
+                    + " = ? AND "
+                    + DATABASE_NAME
+                    + " = ? AND "
+                    + DATABASE_PROPERTY_KEY
+                    + " IN ";
+    static final String DELETE_ALL_DATABASE_PROPERTIES_SQL =
+            "DELETE FROM "
+                    + DATABASE_PROPERTIES_TABLE_NAME
+                    + " WHERE "
+                    + CATALOG_NAME
+                    + " = ? AND "
+                    + DATABASE_NAME
+                    + " = ?";
+    static final String LIST_PROPERTY_DATABASES_SQL =
+            "SELECT DISTINCT "
+                    + DATABASE_NAME
+                    + " FROM "
+                    + DATABASE_PROPERTIES_TABLE_NAME
+                    + " WHERE "
+                    + CATALOG_NAME
+                    + " = ? AND "
+                    + DATABASE_NAME
+                    + " LIKE ?";
+    static final String LIST_ALL_PROPERTY_DATABASES_SQL =
+            "SELECT DISTINCT "
+                    + DATABASE_NAME
+                    + " FROM "
+                    + DATABASE_PROPERTIES_TABLE_NAME
+                    + " WHERE "
+                    + CATALOG_NAME
+                    + " = ?";
+
+    // Distributed locks table
+    static final String DISTRIBUTED_LOCKS_TABLE_NAME = "paimon_distributed_locks";
+    static final String LOCK_NAME = "lock_name";
+    static final String ACQUIRED_AT = "acquired_at";
+
+    static final String CREATE_DISTRIBUTED_LOCK_TABLE_SQL =
+            "CREATE TABLE "
+                    + DISTRIBUTED_LOCKS_TABLE_NAME
+                    + "("
+                    + LOCK_NAME
+                    + " VARCHAR(1000) NOT NULL,"
+                    + ACQUIRED_AT
+                    + " TIMESTAMP NULL DEFAULT NULL,"
+                    + "PRIMARY KEY ("
+                    + LOCK_NAME
+                    + ")"
+                    + ")";
+
+    static final String DISTRIBUTED_LOCK_ACQUIRE_SQL =
```

Comment:
Can you explain this? How to lock?

Reply:
> Can you explain this? How to lock?

@JingsongLi First, create a table for distributed locks and write a data entry to the database table using database.table as the lockId. If successful, the lock acquisition is considered successful. If the primary key conflicts, it is considered a failure